clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractStickyAssignor.java

```diff
@@ -130,19 +146,26 @@ private boolean allSubscriptionsEqual(Set<String> allTopics,
                 for (final TopicPartition tp : memberData.partitions) {
                     // filter out any topics that no longer exist or aren't part of the current subscription
                     if (allTopics.contains(tp.topic())) {
-                        ownedPartitions.add(tp);
+
+                        if (!allPreviousPartitionsToOwner.containsKey(tp)) {
+                            allPreviousPartitionsToOwner.put(tp, consumer);
+                            ownedPartitions.add(tp);
+                        } else {
+                            String otherConsumer = allPreviousPartitionsToOwner.get(tp);
+                            log.warn("Found multiple consumers {} and {} claiming the same TopicPartition {} in the "
```

Comment:
This is fix #2 -- if we somehow still get multiple consumers claiming a partition in the same generation, we have to consider both invalid and remove it from their `ownedPartitions`

Reply:
nit: do you think we should log at ERROR since this is not expected really? Right now we would sort of "hide" such bugs and still be able to proceed silently; I feel we should shouting out such scenarios a bit louder in logs.

Reply:
Good point, yes I would absolutely want/hope a user would report this as a bug. Changed to ERROR